packages/shared/wallet/src/hooks/use-intercom-update.tsx

```diff
@@ -0,0 +1,38 @@
+import { formatNumberAsAbbr } from '@notional-finance/helpers';
+import { useNotionalContext } from '@notional-finance/notionable-hooks';
+import { SupportedNetworks } from '@notional-finance/util';
+import { useIntercom } from 'react-use-intercom';
+import { useEffect } from 'react';
+
+export const useIntercomUpdate = () => {
+  const { update } = useIntercom();
+  const {
+    globalState: { networkAccounts, wallet },
+  } = useNotionalContext();
+  const isReadOnlyAddress = wallet?.isReadOnlyAddress;
+  const selectedAccount = wallet?.selectedAddress;
+
+  useEffect(() => {
+    if (selectedAccount && !isReadOnlyAddress) {
+      const networkBalanceData = {};
+      SupportedNetworks.forEach((network) => {
+        const account = networkAccounts ? networkAccounts[network] : undefined;
+        const usdBalance = account?.accountDefinition?.balances.reduce(
+          (sum, b) => sum + b.toFiat('USD').abs().toFloat(),
+          0
+        );
+        networkBalanceData[network] = usdBalance
+          ? formatNumberAsAbbr(usdBalance)
+          : '';
+      });
+
+      update({
+        userId: selectedAccount,
+        name: selectedAccount,
+        customAttributes: {
```

Comment:
You don't need to destructure here? Can't you just do `customAttributes: networkBalanceData`